nano/node/transport/udp.hpp

```diff
@@ -69,14 +88,29 @@ namespace transport
 			node_id = node_id_a;
 		}
 
+		nano::transport::transport_type get_type () const
+		{
+			std::lock_guard<std::mutex> lk (channel_mutex);
+			return type;
+		}
+
+		void set_type (nano::transport::transport_type type_a)
+		{
+			std::lock_guard<std::mutex> lk (channel_mutex);
+			type = type_a;
+		}
+
 		unsigned network_version{ nano::protocol_version };
+		std::shared_ptr<nano::socket> socket;
```

Comment:
This is mixing TCP in to a UDP class and I think these should remain separate.

Reply:
Updated